siliconcompiler/core.py

```diff
@@ -1360,20 +1360,36 @@ def _merge_manifest(self, src, job=None, clobber=True, clear=True, check=False,
                 if not key_valid:
                     self.logger.warning(f'Keypath {keylist} is not valid')
             if key_valid and 'default' not in keylist:
-                # update value, handling scalars vs. lists
                 typestr = src.get(*keylist, field='type')
                 for val, step, index in src._getvals(*keylist, return_defvalue=False):
+                    # update value, handling scalars vs. lists
                     if re.match(r'\[', typestr) and not clear:
                         dest.add(*keylist, val, step=step, index=index)
+                        add = True
                     else:
                         dest.set(*keylist, val, step=step, index=index, clobber=clobber)
+                        add = False
+
+                    # update other pernode fields
+                    # TODO: only update these if clobber is successful
```

Comment:
What is missing to complete this TODO?

Reply:
It's actually a little tricky, I opened a ticket SC-187 to track with a test case and a couple proposed solutions.